doc/controller/main_controller_class.rst

```diff
@@ -2,10 +2,15 @@
 
 Network controller class
 =====================================
-The network controller class is hte basic controller class. It stores lists of the controllable
-assets. The method run_time_step is used to get the controller settings for the given time step.
-For developers if you want to implement a new controller a class can be created with its own logic
-as long as there is a method run_time_step that returns the controller settings.
+The network controller class is the basic controller class. It stores lists of the controllable
```

Comment:
The NetworkController class serves as the base controller, managing lists of controllable assets. The update_setpoints method retrieves controller settings for a specific time step.

To create a custom controller, implement a new class with your desired logic and inherit from the NetworkControllerAbstract class.

Reply:
done